source/grid/tria_accessor.cc

```diff
@@ -2612,14 +2600,12 @@ CellAccessor<dim, spacedim>::periodic_neighbor(const unsigned int i_face) const
    */
   AssertIndexRange(i_face, this->n_faces());
   using cell_iterator = TriaIterator<CellAccessor<dim, spacedim>>;
-  cell_iterator my_it(*this);
+  cell_iterator current_cell(*this);
+
+  const auto &my_face_pair =
+    this->tria->periodic_face_map.find(std::make_pair(current_cell, i_face));
```

Comment:
`std::map::find` returns an iterator, not a reference. Just make it a non-reference `auto`.